manifests/candlepin.pp

```diff
@@ -31,8 +31,11 @@
   Boolean $manage_db = true,
 ) {
   include certs
+  include certs::foreman
```

Comment:
I don't quite like that this pulls in Foreman certificates, even though they may not be used at all (this class is still intended to be used standalone on a server without Katello).

Instead, I'd suggest to make add `$artemis_client_dn` as a class parameter here. Then in `init.pp` you would pass it. For that to work you need to swap the order of `katello::candlepin` and `katello::application` (due to compile time dependencies, no actual chaining).

You could set `$artemis_client_dn` (as you calculate it below) in `application.pp` and reuse it in `init.pp`. That would signal the relationship.

Reply:
Hrmm, testing that idea out I am running into cyclic dependency issues:

```
        Evaluation Error: Error while evaluating a Resource Statement, Duplicate declaration: Class[Certs::Candlepin] is already declared; cannot redeclare (file: /home/ehelms/workspace/upstream/installer/puppet-katello/spec/fixtures/modules/katello/manifests/candlepin.pp, line: 37) (file: /home/ehelms/workspace/upstream/installer/puppet-katello/spec/fixtures/modules/katello/manifests/candlepin.pp, line: 37, column: 3) on node wareagle.com
```

Thoughts on https://github.com/theforeman/puppet-katello/pull/457#discussion_r983798270 ?